### PR TITLE
[1.11] Add module to cover the Quickstart using S2i

### PR DESCRIPTION
**Pull request by Sgitario**
This quickstart is part of OpenShift quickstarts.
More information in QUARKUS-975

Backport of https://github.com/quarkus-qe/quarkus-openshift-test-suite/pull/272

### CODE REVIEW

File: external-applications/quickstart-using-s2i/src/test/resources/quickstart-using-s2i.yaml [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+    name: quickstart-using-s2i
+  spec:
+    output:
+      to:
+        kind: ImageStreamTag
+        name: quickstart-using-s2i:latest
+    source:
+      git:
+        uri: https://github.com/quarkusio/quarkus-quickstarts
+        ref: "1.11"
```
Comment (Sgitario) [inline external-applications/quickstart-using-s2i/src/test/resources/quickstart-using-s2i.yaml:24]: ! Note: I had to set the branch 1.11 in the remote Quickstart to use it in 1.11

**Comment by Sgitario** [comment]
run tests

**Comment by Sgitario** [comment]
run tests

**Comment by Sgitario** [comment]
@rsvoboda this one is ready to be merged

**Comment by rsvoboda** [comment]
How is ensured that "https://github.com/quarkusio/quarkus-quickstarts/blob/main/getting-started/.s2i/environment" is picked?

**Comment by Sgitario** [comment]
> How is ensured that "https://github.com/quarkusio/quarkus-quickstarts/blob/main/getting-started/.s2i/environment" is picked?

We can see this in the Build pod output. Moreover, it would fail if those settings were not being in use.

**Review by rsvoboda — APPROVED** [review]

**Pull request merged**